core/src/main/java/com/linecorp/armeria/server/websocket/WebSocketHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.websocket;
```

Comment:
Should this be in `common.websocket`? 🤔 e.g.

```
WebSocketClient client = ...;
client.connect("/ws", handler);

// vs.

CompletableFuture<WebSocket> in = client.connect("/ws", out);

// vs.

SomeHolder holder = client.connect("/ws");
holder.in()...
holder.setOut(...)
```

If we're gonna use it for the client-side, we'll have to add an error handler method as well.

I'm fine with leaving this as it is for now since it's an unstable API that needs more user feedback.

Reply:
Yeah, I wasn't sure that we are going to have the same handler for the client side. Let me leave it as is for now and change it when I add the WebSocket client.

Reply:
Yeah, I wasn't sure that we are going to have the same handler for the client side. Let me leave it as is for now and change it when I add the WebSocket client.